src/lune/builtins/luau.rs

```diff
@@ -0,0 +1,129 @@
+                optimization_level: 1,
+                coverage_level: 0,
+                debug_level: 1,
+            },
+            LuaValue::Table(t) => {
+                let optimization_level: Option<u8> = t.get("optimizationLevel")?;
+                let coverage_level: Option<u8> = t.get("coverageLevel")?;
+                let debug_level: Option<u8> = t.get("debugLevel")?;
+
+                Self {
+                    optimization_level: optimization_level.unwrap_or(1),
+                    coverage_level: coverage_level.unwrap_or(0),
+                    debug_level: debug_level.unwrap_or(1),
+                }
+            }
+            _ => {
+                return Err(LuaError::FromLuaConversionError {
+                    from: value.type_name(),
+                    to: "CompileOptions",
+                    message: Some(format!(
+                        "Invalid compile options - expected table, got {}",
+                        value.type_name()
+                    )),
+                })
+            }
+        })
+    }
+}
+
+struct LoadOptions {
+    pub debug_name: String,
+}
+
+impl Default for LoadOptions {
+    fn default() -> Self {
+        Self {
+            debug_name: DEFAULT_DEBUG_NAME.to_string(),
+        }
+    }
+}
+
+impl<'lua> FromLua<'lua> for LoadOptions {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> LuaResult<Self> {
+        Ok(match value {
+            LuaValue::Nil => Self {
+                debug_name: DEFAULT_DEBUG_NAME.to_string(),
+            },
+            LuaValue::Table(t) => {
+                let debug_name: Option<String> = t.get("debugName")?;
+
+                Self {
+                    debug_name: debug_name.unwrap_or(DEFAULT_DEBUG_NAME.to_string()),
+                }
+            }
+            LuaValue::String(s) => Self {
+                debug_name: s.to_str()?.to_string(),
+            },
+            _ => {
+                return Err(LuaError::FromLuaConversionError {
+                    from: value.type_name(),
+                    to: "LoadOptions",
+                    message: Some(format!(
+                        "Invalid compile options - expected string or table, got {}",
+                        value.type_name()
+                    )),
+                })
+            }
+        })
+    }
+}
+
+pub fn create(lua: &'static Lua) -> LuaResult<LuaTable> {
+    TableBuilder::new(lua)?
+        .with_function("compile", compile_source)?
+        .with_function("load", load_source)?
+        .build_readonly()
+}
+
+fn compile_source<'lua>(
+    lua: &'lua Lua,
+    (source, options): (LuaString<'lua>, Option<CompileOptions>),
+) -> LuaResult<LuaString<'lua>> {
+    let _options = options.unwrap_or_default();
+    let source_bytecode_bytes = LuaCompiler::default()
+        .set_optimization_level(_options.optimization_level)
+        .set_coverage_level(_options.coverage_level)
+        .set_debug_level(_options.debug_level)
```

Comment:
```suggestion
    (source, options): (LuaString<'lua>, CompileOptions),
) -> LuaResult<LuaString<'lua>> {
    let source_bytecode_bytes = LuaCompiler::default()
        .set_optimization_level(options.optimization_level)
        .set_coverage_level(options.coverage_level)
        .set_debug_level(options.debug_level)
```

Since the `FromLua` implementation matches on `nil`, we don't need to make this an `Option`